lib/oneview-sdk/resource/api200/logical_switch.rb

```diff
@@ -46,12 +46,13 @@ def create
         self
       end
 
-      # Updates this object using the data that exists on OneView
-      # @note Will overwrite any data that differs from OneView
-      # @return [Resource] self
-      def refresh
+      # Reclaims the top-of-rack switches in a logical switch.
+      # @return [OneviewSDK::LogicalSwitch] self
+      def refresh_state!
```

Comment:
Regarding the method implementation, I took a look in the API and I don't think the `refresh/`endpoint does what this method think it does.

I mean, the response **does not returns the Logical Switch**, it reclaims the associated ToR switches... So, it won't make sense using the `set_all` at all.

Also, since it does not changes anything in the resource, you will probably have to remove the **!** from the method signature